akka-cluster-typed/src/test/scala/docs/akka/cluster/typed/DistributedPubSubExample.scala

```diff
@@ -0,0 +1,356 @@
+  import Ontology._
+
+  def apply(dt: DataType, mediator: akka.actor.ActorRef): Behavior[DataEvent] = {
+    //#destination
+    Behaviors.setup { context =>
+      // register to the path
+      import akka.actor.typed.scaladsl.adapter._
+      mediator ! DistributedPubSubMediator.Put(context.self.toClassic)
+
+      idle(dt, mediator)
+    }
+    //#destination
+  }
+
+  private def idle(dt: DataType, mediator: akka.actor.ActorRef): Behavior[DataEvent] =
+    Behaviors.setup { context =>
+      Behaviors.receiveMessage {
+        case StartIngestion(_, _) =>
+          context.log.info("Processing events for {} storing.", dt.key)
+          mediator ! DistributedPubSubMediator.Publish(
+            IngestionTopic,
+            IngestionStarted(dt.key, context.self.path.toStringWithoutAddress))
+          active(dt, mediator)
+
+        case _ =>
+          idle(dt, mediator)
+      }
+    }
+
+  /** Would normally be typed more specifically. */
+  private def active(dt: DataType, mediator: akka.actor.ActorRef): Behavior[DataEvent] =
+    //#publisher
+    Behaviors
+      .supervise(Behaviors.receiveMessagePartial[DataEvent] {
+        case e: DataEnvelope if e.key == dt.key =>
+          // fanout to:
+          // validate, slice, dice, re-route, store raw to blob, store pre-aggregated/timeseries to Cs*, etc.
+          Behaviors.same
+
+        case StopIngestion(k) if k == dt.key =>
+          mediator ! DistributedPubSubMediator.Publish(IngestionTopic, IngestionStopped(dt.key))
+          // cleanup and graceful shutdown
+          Behaviors.stopped
+      })
+      .onFailure[DeathPactException](SupervisorStrategy.restart)
+  //#publisher
+
+}
+
+object Subscriber {
+  import Ontology._
+
+  def apply(key: DataKey, mediator: akka.actor.ActorRef): Behavior[DataEvent] = {
+
+    //#subscriber
+    Behaviors
+      .setup[DataEvent] { context =>
+        import akka.actor.typed.scaladsl.adapter._
+
+        mediator ! DistributedPubSubMediator.Subscribe(RegistrationTopic, context.self.toClassic)
+        mediator ! DistributedPubSubMediator.Subscribe(IngestionTopic, context.self.toClassic)
+
+        Behaviors.receiveMessagePartial {
+          case dt: DataType if dt.key == key =>
+            // do some capacity planning
+            // allocate some shards
+            // provision a source and sink
+            // start a new ML stream...it's a data platform wonderland
+            wonderland()
+
+          case IngestionStarted(k, path) if k == key =>
+            //#send
+            // simulate data sent from various sources:
+            (1 to 100).foreach { n =>
+              mediator ! DistributedPubSubMediator.Send(
+                path,
+                msg = DataEnvelope(key, s"hello-$key-$n"),
+                localAffinity = true)
+            }
+            //#send
+            continue(key, mediator)
+
+          case _: IngestionStopped =>
+            Behaviors.stopped
+
+        }
+      }
+      .narrow[DataEvent]
```

Comment:
is this needed? if compiler don't like it the type hint can be added to `Behaviors.receiveMessagePartial[DataEvent]`

Reply:
Was but I changed it, so yes now not needed, thanks for the catch.